### PR TITLE
[ART-2974] Prevent unnecessary clone to determine RPM package name

### PR DESCRIPTION
**Pull request by jupierce**
https://issues.redhat.com/browse/ART-2974

### CODE REVIEW

File: doozerlib/rpmcfg.py [hunk 1]
```diff
@@ -233,6 +234,25 @@ def get_package_name(self, default=-1):
...
+        in the RPM's spec file.
+        :param default: If specified, this value will be returned if package name could not be found.
+        :return: The package name - Otherwise, the specified default.
+                If default is not passed in, an exception will be raised if the package name can't be found.
+        """
+        package_name = self.config.name  # This should be the package name for the RPM
+        if package_name:
+            return package_name
+
+        if default != -1:
```
Comment (sosiouxme) [inline doozerlib/rpmcfg.py:251]: this looks weird / non-pythonic... default ought to be `None` and this
```suggestion
        if default is not None:
```
but i guess this is everywhere and we can fix that some other time when it annoys me :)

**Review by sosiouxme — APPROVED** [review]

File: doozerlib/rpmcfg.py [hunk 2]
```diff
@@ -233,6 +234,25 @@ def get_package_name(self, default=-1):
...
+            return default
+
+        raise IOError(f'Missing package name in RPM doozer metadata')
```
Comment (sosiouxme) [inline doozerlib/rpmcfg.py:254]: flake8 doesn't like this being an f-string.
Might be helpful to indicate which distgit has this problem though?

**Review by vfreex — APPROVED** [review]
Unit tests need to be fixed.

**Review by vfreex — CHANGES_REQUESTED** [review]
Unit tests need to be fixed.

**Comment by jupierce** [comment]
Fixed the test.

**Pull request merged**